### ISSUE TITLE
The 'when_present' and 'present?' methods behave strangely when chained

### ISSUE THREAD

**Issue by Pudlovich** [watir/watir#432]
In my test suite I managed to get an error:
```
Failure/Error: expect(b.div(class: 'fc-event').when_present.present?).to be true
       expected true
            got false
```
And this is not the kind of error that I expected to see, should something go wrong.
Is there some kind of timing issue between the methods, or do they somehow evaluate presence differently?
Using 'when_present.exists?' instead seems to do the trick.

**Comment by titusfortner** [comment]
This is because of this issue: https://github.com/watir/watir-webdriver/pull/216

It looks like @jarmo's suggestion essentially bypasses the when_present call when you are calling present? on the Decorator.

**Comment by jarmo** [comment]
It's always nice to read things from 3 years ago and be surprised of what I've written :)

Anyway, I agree that this behavior seems strange, but so does the assertion/spec itself. Reading the spec it seems surprising that there's an assertion for element's `present?` after it has been waited for being present as if you don't trust the framework here (hehe).

@titusfortner seems to be right here because as I understand then `method_missing` is not called because delegator method is called directly, thus call to `wait_until` is skipped.

The really correct solution would be to destroy ActiveSupport in it's entirety, but we all know that this will not happen...

Having that in mind, maybe something like this would work:
```ruby
class WhenPresentDecorator < BaseDecorator
    def present?
      Watir::Wait.until(@timeout, @message) { wait_until }
      true
    rescue Watir::Wait::TimeoutError
      false
    end

    private

    def wait_until
      @element.present?
    end
end
```

**Comment by titusfortner** [comment]
I know exactly what my 3 year old comments often look like. :)

I think this code is right. I pushed it to a branch to make sure Travis likes it. @jarmo do you want to do a PR with that code, or would you rather I commit it?

**Comment by jarmo** [comment]
Feel free to commit yourself :) Just don't forget to add spec as well ;)

**Comment by titusfortner** [comment]
Merged: 9aa1ddd8f75dc0ceb4536a167e4596c56d56c58d
